[2d] remove ShardedTensor from fsdp extension

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107487
* #107473
* __->__ #107472

2D Parallel won't use ShardedTensor, and it causes headable for dynamo
to recoginize it, removing it from the runtime flatten/unflatten path

wanchaol committed:
[2d] remove ShardedTensor from fsdp extension

2D Parallel won't use ShardedTensor, and it causes headable for dynamo
to recoginize it, removing it from the runtime flatten/unflatten path

[ghstack-poisoned]

fduwjj commented:
I am not sure if this would break the current checkpoint or not. If not, I think it would be fine.

wz337 commented:
> I am not sure if this would break the current checkpoint or not. If not, I think it would be fine.

I think it should be fine, as it won't affect the 1D FSDP code path, and we removed the 2D test (with sharded tensor). Our unit tests should be sufficient to confirm whether it will break current checkpoint.

wz337 commented:
> I am not sure if this would break the current checkpoint or not. If not, I think it would be fine.

I think it should be fine, as it won't affect the 1D FSDP code path, and we removed the 2D test (with sharded tensor). Our unit tests should be sufficient to confirm whether it will break current checkpoint.

wanchaol committed:
Update on "[2d] remove ShardedTensor from fsdp extension"


2D Parallel won't use ShardedTensor, and it causes headable for dynamo
to recoginize it, removing it from the runtime flatten/unflatten path

[ghstack-poisoned]

wanchaol commented:
@fduwjj I think the fsdp checkpoint uses `_chunk_tensor` which I didn't delete that, the `flatten_tensor` and `unflatten_tensor` only used in fsdp runtime so I think we should be fine.

wanchaol commented:
@pytorchbot merge -i

huydhn commented:
@pytorchbot drci
